src/elements/content-sidebar/activity-feed/task-new/Assignees.js

```diff
@@ -0,0 +1,113 @@
+// @flow strict
+import * as React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import type { InjectIntlProvidedProps } from 'react-intl';
+import { Flyout, Overlay } from '../../../../components/flyout';
+import Tooltip from '../../../../components/tooltip';
+import messages from '../../../common/messages';
+import AssigneeStatus from './AssigneeStatus';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_COMPLETED, TASK_NEW_NOT_STARTED } from '../../../../constants';
+
+const MAX_AVATARS = 3;
+
+type Props = {|
+    assignees: TaskAssigneeCollection,
+    getAvatarUrl: GetAvatarUrlCallback,
+    maxAvatars?: number,
+    ...InjectIntlProvidedProps,
+|};
+
+const statusMessages = {
+    [TASK_NEW_APPROVED]: <FormattedMessage {...messages.tasksFeedStatusAccepted} />,
+    [TASK_NEW_REJECTED]: <FormattedMessage {...messages.tasksFeedStatusRejected} />,
+    [TASK_NEW_COMPLETED]: <FormattedMessage {...messages.tasksFeedStatusCompleted} />,
+    [TASK_NEW_NOT_STARTED]: <FormattedMessage {...messages.tasksFeedStatusNotStarted} />,
+};
+
+const AssigneeTooltipLabel = React.memo(({ user, status }) => {
```

Comment:
Can we maintain one component per file? Having multiple in one makes testing and reuse more difficult.

Reply:
Thoughts on this, @alexkrolick?

Reply:
I originally had this as just a helper function, but thought adding a React.memo would be a cheap optimization, so it became a component. I've found it difficult to refactor components (like Comment) when they are broken up while still being tightly coupled to the parent. I'll take a pass at re-abstracting the Task and Comment cards after this.